### PR TITLE
feat(unwrapOr): Implement the unwrapOr method.

### PR DESCRIPTION
**Pull request by pawndev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -315,6 +315,30 @@ parseResult.isErr() // true
 
 ---
 
+### `Result.unwrapOr` (method)
+
+Unwrap the `Ok` value, or return de default if there is an `Err`
```
Comment (paduc) [inline README.md:320]: typo here on `or return _de_ default`

**Comment by paduc** [comment]
This would definitely be a welcome addition for me !

File: README.md (hunk 1 above)
Comment (pawndev) [reply README.md:320]: 🙈 I will change this. Thanks !

Comment (paduc) [reply README.md:320]: Simply unacceptable. 😉

**Comment by supermacro** [comment]
Awesome! Very exciting addition to `neverthrow`. 

Thank you @pawndev for the contribution, and thanks @paduc for the second pair of eyes!

**Comment by supermacro** [comment]
It's live!

https://github.com/supermacro/neverthrow/releases/tag/v2.6.0

I'm thinking it would make sense to also add this method to `ResultAsync`.

Do either of you have time to implement this?

```
ResultAsync<T, E>.unwrapOr(defaultVal: T) => Promise<T>
```

**Comment by paduc** [comment]
@supermacro I thought about it but came to the conclusion that it is probably not necessary. When using a `ResultAsync` unwrapping always implies waiting for the promise.

```typescript
const resultAsync: ResultAsync<number, Error> = getMyNumber();

const myNumber  = (await resultAsync).unwrapOr(10);
```

Ok so we could avoid the parens with a `ResultAsync.unwrapOr` => `const myNumber = await resultAsync.unwrapOr(10);` but is it worth it ?

**Comment by pawndev** [comment]
@supermacro awesome ! Thanks for merging/tagging it so fast !

@paduc I agree with you, but I'll take a look to see if it's worth to implement it

**Comment by supermacro** [comment]
The reason I like the idea of having it for `ResultAsync` is because I see `ResultAsync` as a **near-exact** copy of `Result` ... but just asynchronous. 

So in my "vision" for `neverthrow`, I always want to have the same methods on `ResultAsync` as there are for `Result`. That way, there is zero (or close to zero) mental overhead when context-switching between `Result`s and `ResultAsync`s.

**Comment by paduc** [comment]
Great vision indeed ! I just opened a new PR with this small addition.

**Pull request merged**